file_catalog/mongo.py

```diff
@@ -192,3 +193,25 @@ def create_snapshot(self, metadata):
     @run_on_executor
     def get_snapshot(self, filters):
         return self.client.snapshots.find_one(filters, {'_id':False})
+
+    @run_on_executor
+    def append_distinct_elements_to_file(self, uuid, metadata):
+        """Append distinct elements to arrays within a file document."""
+        # build the query to update the file document
+        update_query = {"$addToSet": {}}
+        for key in metadata.keys():
```

Comment:
Suggest `for key in metadata`.  There shouldn't be a need to call `.keys()`.

Reply:
Ahhh yes, good call. Should be fixed now.